jwt/src/main/java/me/uport/sdk/jwt/JWTTools.kt

```diff
@@ -196,59 +189,99 @@ class JWTTools(
             throw InvalidJWTException("JWT has expired: exp: ${payload.exp}")
         }
 
-        val ddo: DIDDocument = UniversalDID.resolve(payload.iss)
+        val publicKeys = resolveAuthenticator(header.alg, payload.iss)
+
+        if (header.alg == JwtHeader.ES256K_R) {
+            if (verifyRecoverableES256K(publicKeys, signatureBytes, token.substringBeforeLast('.').toByteArray())) return payload
+        }
+        else if (header.alg == JwtHeader.ES256K) {
+            if (verifyES256K(publicKeys, signatureBytes)) return payload
+        }
+
+        when (header.alg) {
```

Comment:
This when statement duplicates the if/else code from above

Reply:
My bad, I forgot to delete the previous implementation